source/ui/Components/editor/Editor.qml

```diff
@@ -170,102 +199,488 @@ ScrollView {
                     id: fontMetrics
                     font: textArea.font
                 }
+

```

Comment:
There is quite much code and logic in the Editor.qml file now, which makes it kind of a pain to go through. We should consider refactoring this file and pulling as many components as possible into separate qml files to simplify the editor, if we have time for that (There are probably more important things right now).